checker/src/org/checkerframework/checker/interning/qual/InternedDistinct.java

```diff
@@ -0,0 +1,24 @@
+package org.checkerframework.checker.interning.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.SubtypeOf;
+
+/**
+ * Indicates that no other value is {@code equals()} to the given value. Therefore, it is correct to
+ * use == to test an InternedDistinct value for equality against any other value.
+ *
+ * <p>This is a stronger property than {@link Interned}, but a weaker property than every value of a
+ * Java type being interned.
+ *
+ * @see org.checkerframework.checker.interning.InterningChecker
+ * @checker_framework.manual #interning-checker Interning Checker
+ */
+@SubtypeOf(Interned.class)
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
```

Comment:
As this is the new bottom in the interning type hierarchy, the `@DefaultFor(value = {TypeUseLocation.LOWER_BOUND})` should be moved from Interned to here.
It probably also makes sense to make InternedDistinct implicitfor `typeNames = {Void.class}`, instead of leaving that with Interned.

Reply:
Fixed in commit bd5f660